examples/records/records.bal

```diff
@@ -1,52 +1,85 @@
+public function main(string... args) {
+    // This creates an instance of the `Student` record with an empty record literal. The implicit initial value of each
     // field type is assigned to the corresponding field.
-    Person p1 = {};
-    io:println(p1);
+    Student s1 = {};
+    io:println(s1);
 
-    // This creates an instance of the `Person` record with a non-empty record literal.
-    Person p2 = { name: "Jack", age: 10, parent: p1 };
-    io:println(p2);
+    // This creates an instance of the `Student` record with a non-empty record literal.
+    Student s2 = { name: "John Doe", age: 17, grades: {maths: 80, physics: 75, chemistry: 65}};
+    io:println(s2);
 
     // This fetches the value of a specific field in this record. Fields can be accessed using dot (.) or the index operator.
-    io:println(p2.name);
-    io:println(p2["name"]);
+    io:println(s2.name);
+    io:println(s2["name"]);
 
-    // This fetches the field of a nested record.
-    io:println(p2.parent.age);
+    // This fetches a field of a nested record.
+    io:println(s2.grades.maths);
 
     // This sets the value of a field.
-    p1.name = "Peter";
-    p1.age = 35;
-    io:println(p1);
-    io:println(p2);
+    s1.name = "Peter";
+    s1.age = 19;
+    io:println(s1);
+    io:println(s2);
 
     // This adds an additional field not defined in the record type definition above.
-    p1.occupation = "Ballerina Developer";
-    io:println(p1);
+    // Attempts to add additional fields to a closed record will result in compile errors.
+    // e.g., `s1.grades.ict = 77;`
+    s1.department = "Computer Science";
+    io:println(s1);
 
-    Person[] emps = [p1];
-    // Attempts to add additional fields to a sealed record will result in compile errors.
-    Department dpt = {name:"RnD", employees:emps};
-    io:println(dpt);
+    // Records are iterable and as such, they can be used with foreach loops. The foreach variables are the field name and the value.
+    // The field name's type is `string`. The type of the variable for the value depends on the types of the fields of the record.
+    // If the fields and the rest descriptor are of same type (or if it is a closed record, the rest descriptor does not matter),
+    // the variable will also be of that same type. If not, the variable will be an `any` variable.
+    int total = 0;
+    foreach subj, grade in s2.grades {
+        total += grade;
+        io:println(subj + " : " + grade);
+    }
+
+    io:println("Average grade: " + total/3.0);
+
+    // Since records are iterable, the iterable operations `foreach()`, `map()`, `filter()`, `count()` can also be performed on records.
+    // In addition to these, if all the fields (including rest descriptor, if it is an open record) are either of type `int` or `float`,
+    // the following iterable operations are possible as well: `sum()`, `average()`, `max()`, `min()`.
```

Comment:
would it sound better to say "the iterable operations  `sum()`, `average()`, `max()` and `min()` are also allowed."